corehq/apps/app_manager/app_translations/app_translations.py

```diff
@@ -183,6 +189,40 @@ def process_bulk_app_translation_upload(app, f):
     return msgs
 
 
+def validate_bulk_app_translation_upload(app, workbook, send_email, email):
+    from corehq.apps.app_manager.app_translations.validator import UploadedTranslationsValidator
+    msgs = UploadedTranslationsValidator(app, workbook).compare()
+    if msgs:
+        if send_email:
+            _email_app_translations_discrepancies(msgs, email, app.name)
+            return [(messages.error, _("Issues found. You should receive an email shortly."))]
+        else:
+            message = ["Sheet {}: {}".format(sheet_name, ' '.join(msgs[sheet_name]))
+                       for sheet_name in msgs
+                       if msgs.get(sheet_name)]
+            return [
+                (messages.error, ' '.join(message))
+            ]
+    else:
+        return [(messages.success, "No issues found.")]
+
+
+@task(queue="email_queue")
+def _email_app_translations_discrepancies(msgs, email, app_name):
```

Comment:
I'd use `send_html_email_async` instead; it has retry logic built in and deals with both a text and html version of the email.